design/gitops-cluster-tracers.md

```diff
@@ -0,0 +1,35 @@
+# GitOps Cluster Tracers
+
+KOTS (Admin Console) does not have the ability to detect the currently deployed application version in a GitOps cluster workflow.
+This proposal adds tracers (annotations) to the application pods so that it can be possible to detect that in the future.
+
+## Goals
+
+- Add annotations to application pods to help detect deployments in GitOps clusters in the future
+
+## Non Goals
+
+- Detect the currently deployed application version in a GitOps cluster
+
+## Background
+
+When customers decide to deploy their application via a GitOps workflow, they will probably want to know which version of the app is currently deployed in the cluster, which is currently not possible.
+
+## High-Level Design
+
+Add annotations for both the application slug and sequence to all of the application pods via Kustomize.
+Those annotations will be applied to the downstream manifests before commiting the new changes.
+
+## Detailed Design
+
+When (re)writing the midstream in KOTS, the following will be injected into the kustomization.yaml file:
+
+```
+commonAnnotations:
+  kots.io/app-slug: <app-slug>
+  kots.io/app-sequence: <app-sequence>
+```
+
+both the `<app-slug>` & `<app-sequence>` will be added to the pull/rewrite options before calling kots.pull or kots.rewrite from the KOTS Admin Console.
```

Comment:
I assume these will be optional annotations? How will this work on the first release, IIRC the app-slug is not yet created when we create this archive.

Reply:
the app will be available by then, because we create the app before we create the archive. the sequence will be 0 and the app-slug will also be available